packages/p2p-v2/src/pages/my-profile/screens/MyProfile/MyProfile.tsx

```diff
@@ -14,13 +16,30 @@ const TABS = ['Stats', 'Payment methods', 'Ad details', 'My counterparties'];
 const MyProfile = () => {
     const { isMobile } = useDevice();
     const { queryString, setQueryString } = useQueryString();
+    const { data } = usePoiPoaStatus();
+    const { data: advertiserStats, failureReason, isLoading } = useAdvertiserStats();
+    const { isP2PPoaRequired, isPoaVerified, isPoiVerified } = data || {};
+    const [isNicknameModalOpen, setIsNicknameModalOpen] = useState<boolean | undefined>(false);
 
     const currentTab = queryString.get('tab');
 
+    useEffect(() => {
+        if ((!isP2PPoaRequired || isPoaVerified) && isPoiVerified && !!failureReason) setIsNicknameModalOpen(true);
```

Comment:
nit: what dis condition mean though? Is it just showing the modal when user has verified both poa and poi, then maybe we can re-write it like:
```
const isVerified =  (!isP2PPoaRequired || isPoaVerified) && isPoiVerified
if (isVerified && !!failureReason)
```

Reply:
true, this will be easier to read as well 👍

Reply:
yeah its just to check if the user is poa/poi verified